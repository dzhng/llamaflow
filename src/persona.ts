import { Persona } from './types';

export function buildMessage(persona: Persona): string {
  if (!persona.qualifiers || persona.qualifiers.length === 0) {
    return persona.prompt;
  }

  return `${persona.prompt.trim()}\n\nYou will:\n- ${persona.qualifiers.join('\n- ')}`;
}
