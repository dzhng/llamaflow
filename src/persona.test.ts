import { buildMessage } from './persona';

describe('buildMessage', () => {
  it('Should build system message given a persona', () => {
    expect(
      buildMessage({
        prompt: 'You are a helpful assistant.',
      }),
    ).toMatchSnapshot();
  });

  it('Should build system message given a persona and qualifiers', () => {
    expect(
      buildMessage({
        prompt:
          'You are a smart and honest writer for a TV show about the history of Europe. You will write as concisely and clearly as possible, without factual errors.',
        qualifiers: [
          'Write in an engaging and friendly manner, and never say common misconceptions, outdated information, lies, fiction, myths, or memes.',
          'Include any supporting evidence in that aligns with what you are asked to write.',
          "When writing about any person, explain the person's origin in details",
          "Follow the user's requirements carefully & to the letter.",
        ],
      }),
    ).toMatchSnapshot();
  });
});
