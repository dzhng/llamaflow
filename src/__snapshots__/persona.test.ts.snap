// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buildMessage Should build system message given a persona 1`] = `"You are a helpful assistant."`;

exports[`buildMessage Should build system message given a persona and qualifiers 1`] = `
"You are a smart and honest writer for a TV show about the history of Europe. You will write as concisely and clearly as possible, without factual errors.

You will:
- Write in an engaging and friendly manner, and never say common misconceptions, outdated information, lies, fiction, myths, or memes.
- Include any supporting evidence in that aligns with what you are asked to write.
- When writing about any person, explain the person's origin in details
- Follow the user's requirements carefully & to the letter."
`;
